source/guide/3.2/guide-security.md

```diff
@@ -11,33 +11,230 @@ pageord: 500
+{%note title=Note%}
+In the default configuration, the key used to authenticate a token must be the same as the key used to generate it.
+{%endnote%}
+
+It is possible to implement many other authentication providers, including providers that do not require accessing a
+userstore directly (e.g. oAuth). This is explained later in this document.
+
+### Configuring a Userstore
+Under "userstore_driver" a single userstore is set:
+
+- implementation - the fully qualified name of the module implementing a userstore integration, followed by ":" and the
+class name.
+- properties - a dictionary of arguments required to instantiate the implementing class. The arguments will be passed as
+kwargs to the class' `__init__` method.
+
+The default configuration uses Flask-secuREST's simple userstore, with a list of users in-line:
+
+{% highlight yaml %}
+userstore_driver:
+  implementation: flask_securest.userstores.simple:SimpleUserstore
+  properties:
+    userstore:
+      user1:
+        username: example_user1
+        password: example_password1
+        email: example_user1@your_domain.dom
+      user2:
+        username: example_user2
+        password: example_password2
+        email: example_user2@your_domain.dom
+      user3:
+        username: example_user3
+        password: example_password3
+        email: example_user3@your_domain.dom
+    identifying_attribute: username
+{% endhighlight %}
 
-### setting security on / off
+In the default configuration a userstore is created on the fly, containing the 3 listed users. The identifying
+attribute (by which a user can be identified) is set to "username".
+This is a very simple implementation useful for demonstration. To integrate with a "real" userstore a different
+implementation can be created and installed, as explained later.
 
-### configuring a userstore
 
-### configuring authentication providers
+### Configuring a Token Generator
+In order to enable token generation through the REST "/tokens" endpoint a token generator must be configured:
+
+- implementation - the fully qualified name of the module implementation token generation, followed by ":" and the
+class name.
+- properties - a dictionary of arguments required to instantiate the implementing class. The arguments will be passed as
+kwargs to the class' `__init__` method.
+
+The default configuration uses Flask-secuREST's token module to generate tokens:
+
+{% highlight yaml %}
+auth_token_generator:
+  implementation: flask_securest.authentication_providers.token:TokenAuthenticator
+  properties:
+    secret_key: my_secret
+    expires_in_seconds: 600
+{% endhighlight %}
+
+This configuration uses (as it must) the same secret key for token generation and for token authentication.
+This configuration however includes the additional argument "expires_in_seconds" which limits the lifetime of a token
+to 10 minutes. A token older than 10 minutes will therefore be "expired" and fail the request.
```

Comment:
think this should clear up whether the expire-in-time a /tokens endpoint feature or rather its specific to the default token generator